### PR TITLE
fix #37 and not try to wait for an entry which will never appear

### LINKED ISSUE #37: [Bug] System Ready check fails on older platforms (e.g. Broadcom Sonic v3.5.1)
## Description

Issue #20 brought in a check for system readiness.
This check only works on newer switches running Broadcom SoniC v4.0.0 ++.
So for older software releases we need to only rely on SYSTEM_READY.

### PR DESCRIPTION
**Pull request by Cellebyte**
This should fix older platforms to not wait for `SYSTEM_READY_CORE|SYSTEM_STATE` which is not available.

### CODE REVIEW

**Review by kamalkrbh — APPROVED** [review]

**Review by kamalkrbh — APPROVED** [review]

File: src/sonic_exporter/config/logging.yml [hunk 1]
```diff
@@ -1,13 +1,6 @@
 version: 1
 
 handlers:
-    file:
```
Comment (kamalkrbh) [inline src/sonic_exporter/config/logging.yml:4]: @Cellebyte File based logs are better for debugging, docker logs or consol logs and quickly grow big also hard to send for debugging.
We should keep file based logging.

File: src/sonic_exporter/__init__.py [hunk 2]
```diff
@@ -1,10 +0,0 @@
-import logging
```
Comment (kamalkrbh) [inline src/sonic_exporter/__init__.py:1]: @Cellebyte 
__init__.py is a better place to initiate/config logging like services. 
__init__.py will always be executed if some one uses sonic-exporter in their project deps.

File: src/sonic_exporter/exporter.py [hunk 3]
```diff
@@ -1821,12 +1988,30 @@ def main():
         os.environ.get("SONIC_EXPORTER_PORT", 9101)
     )  # setting port static as 9101. if required map it to someother port of host by editing compose file.
     address = str(os.environ.get("SONIC_EXPORTER_ADDRESS", "localhost"))
-    sonic_collector = SONiCCollector(os.environ.get(
-        "DEVELOPER_MODE", "False").lower() in TRUE_VALUES)
-    _logger.info(
-        "Starting Python exporter server at {}:{}".format(address, port))
+    logging_config_path = os.environ.get(
```
Comment (kamalkrbh) [inline src/sonic_exporter/exporter.py:1991]: @Cellebyte , Related to comment on __init__.py

File: README.md [hunk 4]
```diff
@@ -192,7 +192,7 @@ ssh "admin@${SWITCH}" "docker load -i ${SONIC_EXPORTER_FILE}"
 1. SONIC exporter
 
 ```console
-$ docker run -e SONIC_EXPORTER_ADDRESS="0.0.0.0" --name sonic-exporter --network=host --pid=host --privileged --restart=always -d -v /var/run/redis:/var/run/redis -v /usr/bin/vtysh:/usr/bin/vtysh -v /usr/bin/docker:/usr/bin/docker -v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/ntpq:/usr/bin/ntpq -v /usr/lib/x86_64-linux-gnu/:/usr/lib/x86_64-linux-gnu/  ${SONIC_EXPORTER_IMAGE}
+$ docker run -e SONIC_EXPORTER_ADDRESS="0.0.0.0" --name sonic-exporter --network=host --pid=host --privileged --restart=always -d -v /var/run/redis:/var/run/redis -v /usr/bin/vtysh:/usr/bin/vtysh -v /usr/bin/docker:/usr/bin/docker -v /var/run/docker.sock:/var/run/docker.sock -v /usr/local/bin/sonic-cfggen:/usr/local/bin/sonic-cfggen -v /usr/bin/ntpq:/usr/bin/ntpq ${SONIC_EXPORTER_IMAGE}
```
Comment (kamalkrbh) [inline README.md:195]: @Cellebyte , Similar command should be updated at the begining of readme under section "Install sonic-exporter"

Comment (Cellebyte) [reply README.md:195]: You are right and we still need to investigate the ntp_sync_status. As it is correctly reporting the status.

File: src/sonic_exporter/config/logging.yml (hunk 1 above)
Comment (Cellebyte) [reply src/sonic_exporter/config/logging.yml:4]: We should not do file based logging as it is not correctly handled in logrotate. I changed the rub command to limit the log size.

Comment (kamalkrbh) [reply src/sonic_exporter/config/logging.yml:4]: Thanks @Cellebyte for feedback, can you please tell more about logrotate issue, I will look into it. But file based logging is helpful anyways. :)

Comment (Cellebyte) [reply src/sonic_exporter/config/logging.yml:4]: It is but it then needs to be toggled as if you ran in development mode it wants to write to a location which is not writable by a normal user.

Comment (kamalkrbh) [reply src/sonic_exporter/config/logging.yml:4]: Yes, as logging is customizable, so file based logging can easily be disabled or log file location can be changed as per user need.

BTW, Currently sonic_exporter image is built with default user "root" and sonic_exporter runs as root in the container hence /var/log/ location is writable by the process, Also In developer_mode the same image runs, just with additional flag "-e DEVELOPER_MODE="true", In this case as well sonic_exporter process should be able to write to default /var/log/ location. Or may be I need to know more about your development setup.

Comment (Cellebyte) [reply src/sonic_exporter/config/logging.yml:4]: Thats only true for the container runtime docker.
If you run sonic-exporter as a user process. Which would be the case if you run it in crio or podman then you can't write to such locations.
What we could do is change the log output to a local location when the container is started in development mode.

Comment (kamalkrbh) [reply src/sonic_exporter/config/logging.yml:4]: @Cellebyte , Yes, That can be a solution. 
Will resolve the conversation and close pull request on receving updates for above said solution. :)

**Pull request merged**